File: mumal_hr/mumal_hr/report/employee_score_report/employee_score_report.py
import frappe

def execute(filters=None):
    # Define columns for the report
    columns = [
        {"label": "Employee ID", "fieldname": "employee", "fieldtype": "Link", "options": "Employee"},
        {"label": "Employee Name", "fieldname": "employee_name", "fieldtype": "Data"},
        {"label": "Department", "fieldname": "department", "fieldtype": "Link", "options": "Department"},
        {"label": "Designation", "fieldname": "designation", "fieldtype": "Link", "options": "Designation"},
        {"label": "Mobile No", "fieldname": "mobile_no", "fieldtype": "Data"}
    ]

    # Initialize data as an empty list
    data = []

    # Base SQL query
    sql_query = """
        SELECT DISTINCT
            s.employee,
            s.employee_name,
            e.department,
            e.designation,
            e.cell_number AS mobile_no
        FROM `tabSalary Slip` s
        LEFT JOIN `tabEmployee` e ON s.employee = e.name
        WHERE e.custom_pf = 1
    """

    # Prepare the condition for filtering
    condition = ""
    params = {}

    if filters:
        company = filters.get("company")
        from_date = filters.get("from_date")
        to_date = filters.get("to_date")

        if company:
            condition += " AND e.company = %(company)s"
            params["company"] = company

        if from_date and to_date:
            condition += " AND s.start_date >= %(from_date)s AND s.end_date <= %(to_date)s"
            params["from_date"] = from_date
            params["to_date"] = to_date

    # Final SQL Query
    final_query = sql_query + condition

    # Execute SQL query
    data = frappe.db.sql(final_query, params, as_dict=True)

    return columns, data

File: mumal_hr/mumal_hr/report/employee_score_report/employee_score_report.js
// Copyright (c) 2024, jignasha@sanskartechnolab.com and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Employee Score Report"] = {
	"filters": [
			{
				"fieldname": "from_date",
				"label": "From Date",
				"fieldtype": "Date",
				"on_change": function() {
					var from_date = frappe.query_report.get_filter_value('from_date');
				
					if (from_date) {
						var toDateObj = frappe.datetime.add_days(from_date, 6);
						frappe.query_report.set_filter_value('to_date', toDateObj);
					}
				},		
				"reqd": 1
			},
			{
				"fieldname": "to_date",
				"label": "To Date",
				"fieldtype": "Date",
				"reqd": 1
			},
			{
				"fieldname": "user",
				"label": "User",
				"fieldtype": "Link",
				"options": "User",
				// "reqd": 1
			}
	],


	onload: function(report) {
        // Add a custom button to duplicate the report data
        report.page.add_inner_button(__("Employee Target Score Summary"), function() {
            // Retrieve filter values from the report
            var filters = report.get_values();

            // Construct the URL for the report view with filter values
            var url = "/app/query-report/Employee Target Score Summary";
            if (filters) {
                var query_params = [];
                Object.keys(filters).forEach(function(key) {
                    query_params.push(key + "=" + encodeURIComponent(filters[key]));
                });
                url += "?" + query_params.join("&");
            }

            // Open the report view URL in the same window
            window.open(url, '_self');
        });
    }
	
};
